sdks/java/core/src/test/java/org/apache/beam/sdk/coders/CommonCoderTest.java

```diff
@@ -0,0 +1,219 @@
+  abstract static class CommonCoderTestSpec {
+    abstract CommonCoder getCoder();
+    abstract @Nullable Boolean getNested();
+    abstract Map<String, Object> getExamples();
+    @JsonCreator
+    static CommonCoderTestSpec create(
+        @JsonProperty("coder") CommonCoder coder,
+        @JsonProperty("nested") @Nullable Boolean nested,
+        @JsonProperty("examples") Map<String, Object> examples) {
+      return new AutoValue_CommonCoderTest_CommonCoderTestSpec(coder, nested, examples);
+    }
+  }
+
+  @AutoValue
+  abstract static class OneCoderTestSpec {
+    abstract CommonCoder getCoder();
+    abstract boolean getNested();
+    @SuppressWarnings("mutable")
+    abstract byte[] getSerialized();
+    abstract Object getValue();
+    static OneCoderTestSpec create(
+        CommonCoder coder, boolean nested, byte[] serialized, Object value) {
+      return new AutoValue_CommonCoderTest_OneCoderTestSpec(coder, nested, serialized, value);
+    }
+  }
+
+  private static List<OneCoderTestSpec> loadStandardCodersSuite() throws IOException {
+    InputStream stream = CommonCoderTest.class.getResourceAsStream(STANDARD_CODERS_YAML_PATH);
+    if (stream == null) {
+      fail("null stream");
+    }
+
+    // Would like to use the InputStream directly with Jackson, but Jackson does not seem to
+    // support streams of multiple entities. Instead, read the entire YAML as a String and split
+    // it up manually, passing each to Jackson.
+    String specString = CharStreams.toString(new InputStreamReader(stream));
+    String[] specs = specString.split("\n---\n");
+    List<OneCoderTestSpec> ret = new LinkedList<>();
+    for (String spec : specs) {
+      CommonCoderTestSpec coderTestSpec = parseSpec(spec);
+      CommonCoder coder = coderTestSpec.getCoder();
+      for (Map.Entry<String, Object> oneTestSpec : coderTestSpec.getExamples().entrySet()) {
+        byte[] serialized = oneTestSpec.getKey().getBytes(StandardCharsets.ISO_8859_1);
+        Object value = oneTestSpec.getValue();
+        if (coderTestSpec.getNested() == null) {
+          // Missing nested means both
+          ret.add(OneCoderTestSpec.create(coder, true, serialized, value));
+          ret.add(OneCoderTestSpec.create(coder, false, serialized, value));
+        } else {
+          ret.add(OneCoderTestSpec.create(coder, coderTestSpec.getNested(), serialized, value));
+        }
+      }
+    }
+    return ImmutableList.copyOf(ret);
+  }
+
+  @Parameters(name = "{1}")
+  public static Iterable<Object[]> data() throws IOException {
+    ImmutableList.Builder<Object[]> ret = ImmutableList.builder();
+    for (OneCoderTestSpec test : loadStandardCodersSuite()) {
+      // Some tools cannot handle Unicode in test names, so omit the problematic value field.
+      String testname = MoreObjects.toStringHelper(OneCoderTestSpec.class)
+          .add("coder", test.getCoder())
+          .add("nested", test.getNested())
+          .add("serialized", test.getSerialized())
```

Comment:
Wondering if it is better to use the index of the example instead of the actual serialized data here.